src/core/thread/meshcop_tlvs.hpp

```diff
@@ -568,13 +568,27 @@ class SteeringDataTlv: public Tlv
      * @retval FALSE  If the TLV does not appear to be well-formed.
      *
      */
-    bool IsValid(void) const { return GetLength() == sizeof(*this) - sizeof(Tlv); }
+    bool IsValid(void) const { return GetLength() <= sizeof(*this) - sizeof(Tlv); }
 
     /**
      * This method sets all bits in the Bloom Filter to zero.
      *
      */
-    void Clear(void) { memset(mSteeringData, 0, sizeof(mSteeringData)); }
+    void Clear(void) { memset(mSteeringData, 0, GetLength()); }
+
+    /**
+     * Ths method sets all bits in the Bloom Filter to one.
+     *
+     */
+    void Set(void) { memset(mSteeringData, 0xff, GetLength()); }
+
+    /**
+     * This method returns the number of bits in the Bloom Filter.
+     *
+     * @returns The number of bits in the Bloom Filter.
+     *
+     */
+    uint8_t GetNumBits(void) const { return GetLength() * 8; }
```

Comment:
One minor question: while length*8 is more readable, would length<<3 be more optimal, or are we counting on the compiler to do that?